### PR TITLE
Enforce UTF-8 encoding on reading and writing files

### PR DESCRIPTION
**Pull request by Jimbo4350**
We were experiencing testing failures on Windows due to GHC 8.10.7's readFile not using UTF-8 by default. This caused golden file comparison errors when Unicode characters were present. For example:

```
478 ┃ testAllErrorMessages_
      479 ┃   :: forall a
      480 ┃    . (HasCallStack, Error a)
      481 ┃   => String
      482 ┃   -- ^ module name
      483 ┃   -> String
      484 ┃   -- ^ type name
      485 ┃   -> [(String, a)]
      486 ┃   -- ^ list of constructor names and values
      487 ┃   -> TestTree
      488 ┃ testAllErrorMessages_ = ErrorMessage.testAllErrorMessages_ goldenFilesPath
          ┃ │ Incorrect error message in golden file
          ┃ │ Reading file: test/cardano-api-golden/files/golden/errors\Cardano.Api.Fees.ScriptExecutionError\ScriptErrorEvaluationFailed.txt
          ┃ │ Golden test failed against the golden file.
          ┃ │ To recreate golden file, run with RECREATE_GOLDEN_FILES=1.
          ┃ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          ┃ │ 37c37
          ┃ │ <      Valid range: (-∞ , +∞)
          ┃ │ ---
          ┃ │ >      Valid range: (-Γê₧ , +Γê₧)
```

### CODE REVIEW

File: src/Hedgehog/Extras/Test/File.hs [hunk 1]
```diff
@@ -170,7 +174,9 @@ openFile filePath mode = GHC.withFrozenCallStack $ do
 readFile :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m String
 readFile filePath = GHC.withFrozenCallStack $ do
   void . H.annotate $ "Reading file: " <> filePath
-  H.evalIO $ IO.readFile filePath
+  liftIO $ IO.withFile filePath IO.ReadMode $ \handle -> do
```
Comment (carbolymer) [inline src/Hedgehog/Extras/Test/File.hs:177]: `evalIO` is needed for proper reporting of IO exceptions
```suggestion
  H.evalIO $ IO.withFile filePath IO.ReadMode $ \handle -> do
```

Why this change only on two functions? I think the change should apply to all functions using string / text when accessing files:
`appendFile`, `textWriteFile`, `textReadFile`, `fileContains`

**Review by carbolymer — CHANGES_REQUESTED** [review]

**Review by newhoggy — APPROVED** [review]

File: src/Hedgehog/Extras/Test/File.hs (hunk 1 above)
Comment (newhoggy) [reply src/Hedgehog/Extras/Test/File.hs:177]: I think you are right.  However, it makes sense to accept this PR as an improvement to the library and do the other fixes in a future PR.

Comment (carbolymer) [reply src/Hedgehog/Extras/Test/File.hs:177]: >do the other fixes in a future PR

Fine but let's do them before making a new release... And the version of hedgehog-extras with this change was tagged already. Can we try to limit technical debt we're introducing?

Or at least put an effort to make technical debt manageable and traceable by creating follow-up tickets/prs?
